src/Datasource/RulesChecker.php

```diff
@@ -314,24 +314,37 @@ protected function _addError($rule, $name, $options)
 
         return function ($entity, $scope) use ($rule, $name, $options) {
             $pass = $rule($entity, $options + $scope);
-            if ($pass === true || empty($options['errorField'])) {
-                return $pass === true;
+            
+            if ($pass === true) {
+                return true;
             }
 
             $message = 'invalid';
+            $save = false;
             if (isset($options['message'])) {
                 $message = $options['message'];
+                $save = true;
             }
             if (is_string($pass)) {
                 $message = $pass;
+                $save = true;
             }
             if ($name) {
                 $message = [$name => $message];
             } else {
                 $message = [$message];
             }
-            $entity->errors($options['errorField'], $message);
-            return $pass === true;
+            
+            $errorField = '_rules';
+            if (isset($options['errorField'])) {
+                $errorField = $options['errorField'];
+                $save = true;
+            }
+            
+            if ($save) {
```

Comment:
Isn't this variable mostly shadowing the presence of a message? It seems like it could be replaced with `isset($message) && !empty($message)`. We would also need to default `$message` to null.